TaskServiceInterview.cs

```diff
@@ -0,0 +1,448 @@
+            if (this.taskLockServant.EmployeeHasLock(task, employeeInitials))
+            {
+                task.ReleaseLock(employeeInitials, this.clock.Current);
+                this.taskChangeNotificationServant.NotifySimpleEvent(task, TaskNotificationItem.OperationType.Update);
+            }
+        }
+
+        public void ReleaseLocks(TaskReleaseLocksRequest request, string employeeInitials)
+        {
+            var tasks = this.taskFinder.FindActiveByLockedByInitials(employeeInitials)
+                .Where(x => request.TaskIds.Contains(x.Id)).ToList();
+            var now = this.clock.Current;
+            foreach (var task in tasks)
+            {
+                task.ReleaseLock(employeeInitials, this.clock.Current);
+                //external call
+                task.Deadline = this.businessCalendarClient.NextBusinessDay(now).Date;
+            }
+
+            this.NotifyAboutTaskChange(tasks, TaskNotificationItem.OperationType.Update);
+        }
+
+        public void ReleaseAllLocks(string employeeInitials)
+        {
+            var tasks = this.taskFinder.FindActiveByLockedByInitials(employeeInitials);
+
+            foreach (var task in tasks)
+            {
+                task.ReleaseLock(employeeInitials, this.clock.Current);
+                //external call
+                task.Deadline = this.businessCalendarClient.NextBusinessDay(this.clock.Current).Date;
+            }
+
+            this.NotifyAboutTaskChange(tasks, TaskNotificationItem.OperationType.Update);
+        }
+
+
+        public void FinishTask(TaskEntity task, string employeeInitials)
+        {
+            this.taskValidator.Validate(task, TaskValidationOperations.Finish);
+            task.Status = TaskStatus.Finished;
+            task.LockedByInitials = null;
+            task.LockedByTimestamp = null;
+
+            var now = this.clock.Current;
+
+            var lockEventTime = this.GetLockTime(task.Id, employeeInitials, now);
+
+            var finishBusinessEventData = new FinishBusinessEventData
+            {
+                TimeToCompleteTicks = (now - lockEventTime).Ticks,
+                DoneAfterDeadline = task.Deadline < this.clock.Current
+            };
+
+            task.SetActionExecutorInformation(ChangeReason.Finish, employeeInitials, now);
+            task.AddBusinessEventFinish(employeeInitials, this.clock.Current, finishBusinessEventData);
+            
+            this.auditLogServant.Log(taskFinished, new { task.Id, employeeInitials });
+        }
+
+        public IEnumerable<TaskListItem> GetUnfinishedTasks(string employeeInitials, bool? limitBySkills)
+        {
+            var tasks = this.taskFinder.FindUnfinishedTasksWithClassification();
+            var taskListItems = this.ConvertToTaskListItems(tasks);
+            var classificationsBasedOnSkills =
+                this.classificationAccessServant.GetClassificationsBasedOnSkills(employeeInitials).ToArray();
+
+            if (limitBySkills.Value)
```

Comment:
osobiscie uzywam if z klamrami, bo wielu moze pomyslec ze return zachodzi przy spelnionym ifie. Jezeli bez bracketow to napisalbym warunek i body w jednej linii